File: apps/mobile/lib/core/cal.dart
import "../core/config.dart";
import "../core/site.dart";

/// Cal.com path without origin, e.g. `nathan-zhao` or `nathan-zhao/30min`.
String getCalPath() {
  final raw = AppConfig.calLink.trim();
  if (raw.isEmpty) {
    return Uri.parse(Site.cal).path.replaceAll(RegExp(r"^/+|/+$"), "");
  }
  return raw
      .replaceFirst(
        RegExp(r"^https?://(www\.)?cal\.com/", caseSensitive: false),
        "",
      )
      .replaceAll(RegExp(r"^/+|/+$"), "");
}

String getCalUrl() {
  final path = getCalPath();
  return path.isEmpty ? Site.cal : "https://cal.com/$path";
}

/// Inline booker URL (keeps UX inside the WebView).
String getCalEmbedUrl() {
  final uri = Uri.parse(getCalUrl());
  return uri
      .replace(
        queryParameters: {
          ...uri.queryParameters,
          "embed": "true",
          "theme": "dark",
          "layout": "month_view",
        },
      )
      .toString();
}

File: apps/desktop/src/lib/cal.ts
import { config } from "@/lib/config";
import { site } from "@/lib/site";

/** Cal.com path without origin, e.g. `nathan-zhao` or `nathan-zhao/30min`. */
export function getCalLink(): string | null {
  const raw = config.calLinkRaw.trim();
  if (!raw) return null;
  const path = raw
    .replace(/^https?:\/\/(www\.)?cal\.com\//i, "")
    .replace(/^\/+|\/+$/g, "");
  return path || null;
}

export function getCalUrl(): string {
  const link = getCalLink();
  return link ? `https://cal.com/${link}` : site.social.cal;
}
